doc/decisions/library_split.md

```diff
@@ -0,0 +1,40 @@
+# Library Split
+
+## Problem
+
+Only libelektra-core is supposed to access private data but this contradicts the goal to keep the library minimal.
+`kdbprivate.h` was too generic, it contained many other parts next to the struct definitions of Key/KeySet.
+
+## Constraints
+
+## Assumptions
+
+- The [C99 standard, section 5.2.4.1](http://www.open-std.org/jtc1/sc22/wg14/) gives following limit:
+  4095 external identifiers in one translation unit
+
+## Considered Alternatives
+
+- keep current situation
+
+## Decision
+
+Also allow `libelektra-operations` library to access private Key/KeySet.
+Put struct definitions of Key/KeySet in a separate header file, which gets
+included by parts that need it
```

Comment:
Continuing from my suggestions above, I would say:
1. It doesn't have to be a single `libelektra-operations` library. If we keep the core minimal, we should probably keep all libraries minimal. Therefore, sometimes two separate libraries make more sense than a single one containing unrelated functions.
2. The header file in question would be `elektra/core-unstable.h`. Anything can include it, but there should be good reason, because it means a breaking change could happen. If a library wants to have a public stable API, but use _unstable_ API, it must mitigate any breaking changes internally.

Reply:
see #4223